tests/unit/s2n_x509_validator_test.c

```diff
@@ -1262,6 +1371,114 @@ int main(int argc, char **argv)
         s2n_x509_validator_wipe(&validator);
         s2n_x509_trust_store_wipe(&trust_store);
     };
+
+    /**
+     * Test OCSP validation at various offsets from update times.
+     * libcrypto ASN1 comparison calculates differences in terms of days and seconds,
+     * so try mocking the system time to a collection of more than day & less
+     * than day differences.
+     * The T's in the below diagram represent test cases that should fail
+     * The F's represent test cases that should succeed
+     *    S2N_ERR_CERT_INVALID                            S2N_ERR_CERT_EXPIRED
+     *          |   |                                        |   |
+     *          v   v                                        v   v
+     *          F   F   T   T                        T   T   F   F
+     *          v   v   v   v                        v   v   v   v
+     * <----------|---|---|----------------------------|---|---|--->
+     *                ^                                    ^
+     *       this update                                 next update
+     *                |---|
+     *                  one day
+     *
+     * If this test is failing make sure that the this_update_timestamp_nanoseconds
+     * matches the actual timestamp of ocsp_response_early_expire.der
+     *
+     * openssl ocsp -respin ocsp_response_early_expire.der -text -noverify | grep "This Update"
+     */
```

Comment:
is this worth putting in the readme too? You mention that this test will break there, but don't mention how to fix it.